plugins/studymaster-web/src/main/java/de/dfki/vsm/xtension/studymasterWeb/WebStudyMasterExecutor.java

```diff
@@ -0,0 +1,306 @@
+import de.dfki.vsm.model.scenescript.ActionFeature;
+import de.dfki.vsm.runtime.activity.AbstractActivity;
+import de.dfki.vsm.runtime.activity.SpeechActivity;
+import de.dfki.vsm.runtime.activity.executor.ActivityExecutor;
+import de.dfki.vsm.runtime.interpreter.value.StringValue;
+import de.dfki.vsm.runtime.project.RunTimeProject;
+import de.dfki.vsm.util.log.LOGConsoleLogger;
+import io.javalin.Javalin;
+import io.javalin.websocket.WsCloseContext;
+import io.javalin.websocket.WsConnectContext;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Objects;
+
+/**
+ * @author Patrick Gebhard, Lenny Händler, Sarah Hoffmann, Fabrizio Nunnari
+ * This plugin uses the Javalin web server to create a remote "Studymaster" connection over websocket
+ * and serve a html/javascript interface for control. The interface source is located in the resources.
+ */
+public class WebStudyMasterExecutor extends ActivityExecutor {
+
+    private static final String sMSG_SEPARATOR = "#";
+    private static final String sMSG_HEADER = "VSMMessage" + sMSG_SEPARATOR;
+
+    /** Project variable set when the "GO" message is received. */
+    private static final String sGO_VAR = "go_var" ;
+    private static final String sGO_VAR_DEFAULT = "go" ;
+
+    /** Project variable holding the result of the user selection (SUBMIT/CANCEL) */
+    private static final String sREQUEST_RESULT_VAR = "request_result_var" ;
+    private static final String sREQUEST_RESULT_VAR_DEFAULT = "request_result" ;
+
+    /** Project variable set when the remote Web GUI connects via websocket. */
+    private static final String sGUI_CONNECTED_VAR = "gui_connected_var" ;
+    private static final String sGUI_CONNECTED_VAR_DEFAULT = "gui_connected" ;
+
+    /** HTTP port */
+    private static final String sJAVALIN_PORT_VAR = "port" ;
+    private static final String sJAVALIN_PORT_VAR_DEFAULT = "8080" ;
+
+
+    /** The singleton logger instance.*/
+    private final LOGConsoleLogger mLogger = LOGConsoleLogger.getInstance();
+    /** The list of websocket connections. Yes, it will be possible to have several GUIs on the same scene flow.*/
+    private final ArrayList<WsConnectContext> mWebsockets = new ArrayList<>();
+    /** The Javalin HTTP server. */
+    private Javalin mHttpServer;
+
+    private String mRequestResultVar ;
+
+    private String mSceneflowGoVar ;
+
+    /** This is the name of a Boolean global project variable that will be bet to _true_ when at least one web app is connected.*/
+    public String mGUIConnectedVar = "";
+
+    /** This is a "cache" of the last request sent to all connected GUIs.
+     * The string contains the message already formatted to the communication protocol.
+     * It will be used to send again the request to GUIs connecting after the REQUEST action was issued.
+     * If null, no pending request is active.
+     */
+    private String mLastRequestMessage = null;
```

Comment:
Why is this String explicitly initialized, while the others are not? (code style consistency)